### PR TITLE
DOOM II: implement new game

### PR DESCRIPTION
**Pull request by Daivuk**
## What is this fixing or adding?

New game.

The client is using a modified version of crispy-doom. You can find it here: https://github.com/Daivuk/apdoom

## How was this tested?

Community tested since July 23rd.

## Notes to reviewers
The python code is all generated, except for Options and __init__. It is using the same generation tool as DOOM 1993. So the locations/items/regions/rules are on the same pattern. If something is required to be changed in those files, then the change will be applied to both games.

## If this makes graphical changes, please attach screenshots.
image
image
image
image
image

### CODE REVIEW

File: worlds/doom_ii/docs/en_DOOM II.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Guns, keycards, and level unlocks have been randomized. Typically, you will end up playing different levels out of order to find your keycards and level unlocks and eventually complete your game.
+
+Maps can be selected on a level select screen. You can exit a level at any time by visiting the hub station at the beginning of each level. The state of each level is saved and restored upon re-entering the level.
+
+## What is the goal?
+
+The goal is to complete every levels.
+
+## What is a "check" in The DOOM II?
```
Comment (ScipioWright) [inline worlds/doom_ii/docs/en_DOOM II.md:17]:
```suggestion

The goal is to complete every level.

## What is a "check" in DOOM II?
```

File: worlds/doom_ii/__init__.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            "Box of bullets",
+            "Box of rockets",
+            "Box of shotgun shells",
+            "Energy cell pack"
+        ])
+
+    def create_ratioed_items(self, item_name: str, itempool: List[DOOM2Item]):
+        remaining_loc = self.location_count - len(itempool)
+
+        # Was balanced for 3 episodes (We added 4th episode, but keep same ratio)
```
Comment (ScipioWright) [inline worlds/doom_ii/__init__.py:223]:
```suggestion

```

Probably just copied over from the DOOM 1 code (very understandable).

**Review by ScipioWright — APPROVED** [review]
I feel like I've reviewed this before...

Jokes aside, looks as good as the first DOOM. Just some small suggestions for the docs, and a comment that probably doesn't belong, and that's all. I have some other thoughts on goals, but I'll message you on discord regarding those, since those shouldn't really be part of this review imo.

**Comment by Daivuk** [comment]
Closing this, as I need more time. We want to make this game episodic. It would be too big of a change later.

**Pull request closed without merging**